make Property an annotation, instead of adding new annotaiton Field

dbu opened the pull request:
see https://github.com/doctrine/phpcr-odm-documentation/pull/74

we need no BC for `@Field` as it was never released.

while doing this, i noticed that in yml mapping, the list of properties is called `fields` which would be an argument in favour of using `@Field` after all. but `@Property` seems more natural for the annotations for me.

dbu commented on lib/Doctrine/ODM/PHPCR/Mapping/Annotations/TranslatedProperty.php:
```diff
@@ -18,13 +18,9 @@
+ * @deprecated Use Property instead
  */
-final class Field extends TranslatableProperty
+class TranslatedProperty extends Property
```
its just git thinking that this is a rename. i delete Field and rename TranslatedProperty to Property, but add this empty class for BC in case some custom code was typechecking TranslatedProperty for something.

dbu commented on lib/Doctrine/ODM/PHPCR/Mapping/Annotations/TranslatableProperty.php:
```diff
@@ -19,12 +19,8 @@
- * Base class for all the translatable properties (i.e. every property but Uuid and Version)
+ * @deprecated Use Property instead
  */
 class TranslatableProperty extends Property
```
keeping for BC in case somebody extended it or has custom code that typechecks.

dbu commented on lib/Doctrine/ODM/PHPCR/Mapping/Annotations/Uuid.php:
```diff
@@ -25,7 +25,7 @@
  * @Target("PROPERTY")
  */
-final class Uuid extends Property
+final class Uuid extends BaseProperty
```
if somebody was typechcking Property in custom code handling annotations, this is a BC break. do we need to worry?

lsmith77 replied:
i would say no

lsmith77 commented:
Changelog also needs to be updated

lsmith77 commented:
tests failing due to
```PHP Fatal error:  Class 'Doctrine\Tests\ODM\PHPCR\Mapping\Model\PropertyMappingObject' not found in /home/travis/build/doctrine/phpcr-odm/tests/Doctrine/Tests/ODM/PHPCR/Mapping/AbstractMappingDriverTest.php on line 22```

dbu commented:
ups. fixed, and tests where green. now updated the changelog.

@dantleech @WouterJ @lsmith77 ok with going like this? if so, i will update the doc as well.

lsmith77 commented:
+1 from me

WouterJ commented:
I'm -1 on this change, but as it's 3 vs 1, I think I have no option...

It's inconsistent with yaml, mongodb odm and couchdb odm. Also, it's a BC break for people having a custom annotation extending from Property (it know suddently becomes translatable, while the function of Property was to have something that's not translatable).

dbu commented:
oh, so mongodb and couchdb use Field? that would indeed be a strong argument in favor of using Field here as well. 

loosk like: http://doctrine-couchdb.readthedocs.org/en/latest/reference/basic-mapping.html#field-mapping

so i am :-1: on my own PR :-/

lsmith77 commented:
alright .. then i will reactivate my docs PR to remove ``@Property``

lsmith77 commented:
merged https://github.com/doctrine/phpcr-odm-documentation/pull/75
